LayerClient/LayerClient.py

```diff
@@ -0,0 +1,509 @@
+
+
+class BaseLayerResponse:
+    """
+    Base class for several of the datatypes returned by Layer - if it returns
+    an ID and a URL, it should extend this class in order to get UUID parsing.
+    """
+
+    def uuid(self):
+        """
+        When we get a conversation response from the server, it doesn't include
+        the raw UUID, only a URI of the form:
+            layer:///conversations/f3cc7b32-3c92-11e4-baad-164230d1df67
+        which must be parsed to retrieve the UUID for easy reference.
+        """
+        uri = urlparse(self.id)
+        path_parts = uri.path.split('/')
+        if len(path_parts) == 3 and len(path_parts[2]) == 36:
+            return path_parts[2]
+
+        return None
+
+
+class PlatformClient(object):
+    """
+    Client to the server-side layer API
+    """
+
+    def __init__(self, app_uuid, bearer_token):
+        """
+        Create a new Layer platform client.
+
+        Parameters:
+        - `app_uuid`: The UUID for the application
+        - `bearer_token`: A Layer authorization token, as generated by the
+            Developer dashboard
+        """
+        self.app_uuid = app_uuid
+        self.bearer_token = bearer_token
+
+    def _get_layer_headers(self):
+        """
+        Convenience method for retrieving the default set of authenticated
+        headers.
+
+        Return: The headers required to authorize ourselves with the Layer
+        platform API.
+        """
+        return {
+            'Accept': 'application/vnd.layer+json; version=1.0',
+            'Authorization': 'Bearer ' + self.bearer_token,
+            'Content-Type': 'application/json'
+        }
+
+    def _get_layer_uri(self, *suffixes):
+        """
+        Used for building Layer URIs for different API endpoints.
+
+        Parameter`suffixes`: An array of strings, which will be joined as the
+            end portion of the URI body.
+
+        Return: A complete URI for an endpoint with optional arguments
+        """
+        suffix_string = '/'.join(suffixes) if suffixes else ''
+        return 'https://api.layer.com/apps/{app_id}/{suffix}'.format(
+            app_id=self.app_uuid,
+            suffix=suffix_string,
+        )
+
+    def _raw_request(self, method, url, data=None):
+        """
+        Actually make a call to the Layer API.
+        If the response does not come back as valid, raises a
+        LayerPlatformException with the error data from Layer.
+
+        Parameters:
+        - `method`: The HTTP method to use
+        - `url`: The target URL
+        - `data`: Optional post body. Must be json encodable.
+
+        Return: Raw JSON doc of the Layer API response
+
+        Exception: `LayerPlatformException` if the API returns non-OK response
+        """
+        result = requests.request(
+            method,
+            url,
+            headers=self._get_layer_headers(),
+            data=(json.dumps(data) if data else None)
+        )
+
+        if not result.ok:
```

Comment:
I would flip this around:
```
if result.ok:
    return result.json()
```

then handle the error cases, because either way it will throw something